src/codegen/codegen_cpp_visitor.hpp

```diff
@@ -1890,12 +1890,15 @@ void CodegenCppVisitor::print_vector_elements(const std::vector<T>& elements,
 template <typename T>
 bool has_parameter_of_name(const T& node, const std::string& name) {
     auto parameters = node->get_parameters();
-    for (const auto& parameter: parameters) {
-        if (parameter->get_node_name() == name) {
-            return true;
+    struct ParameterHasName {
+        const std::string paramater_name;
+        explicit ParameterHasName(const std::string& name)
+            : paramater_name(name) {}
+        bool operator()(const decltype(*parameters.begin()) arg) const {
+            return arg->get_node_name() == paramater_name;
         }
-    }
-    return false;
+    };
+    return std::any_of(parameters.begin(), parameters.end(), ParameterHasName(name));
```

Comment:
Is it possible to use a lambda?
```
    return std::any_of(parameters.begin(), parameters.end(), [&name](const auto& arg) { return arg->get_node_name() == name; });
```

Reply:
Good point :+1: I thought that this function didn't except lambdas in the beginning